packages/block-editor/src/store/private-selectors.js

```diff
@@ -684,3 +686,66 @@ export function getClosestAllowedInsertionPointForPattern(
 export function getInsertionPoint( state ) {
 	return state.insertionPoint;
 }
+
+/**
+ * Retrieves the editing mode of a pattern block and its children.
+ *
+ * @param {Object} state    Global application state.
+ * @param {string} clientId The block client ID.
+ *
+ * @return {string} The block editing mode.
+ */
+export function getPatternBlockEditingMode( state, clientId ) {
```

Comment:
Why is this a separate selector not part of the more generic `getBlockEditingMode`?

Reply:
I thought it might be more readable. I don't have a strong preference though, I've pushed a commit that makes it one selector.

Reply:
Ah, making it separate for clarity like an internal function is fine but the problem with it exposed as a separate selector is the fact that the "framework" when it needs to know the block's editing mode has to switch between two different selectors.